datafusion/functions/src/unicode/character_length.rs

```diff
@@ -136,56 +136,31 @@ where
     // string is ASCII only is relatively cheap.
     // If strings are ASCII only, count bytes instead.
     let is_array_ascii_only = array.is_ascii();
-    let array = if array.null_count() == 0 {
+    let nulls = array.nulls().cloned();
```

Comment:
Is the idea to remove the no-nulls optimization because it doesn't make things faster?